lib/solr_configurator.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'zip'
+
+class SolrConfigurator
+  def initialize
+    connection
+  end
+
+  def zip_file
+    File.open(tempfile)
+  end
+
+  def configset_exists?
+    resp = @connection.get('/solr/admin/configs?action=LIST&omitHeader=true')
```

Comment:
Faraday can accept a hash argument to build the URL parameters for you. This has an advantage of making sure that everything is properly URL escaped. Ditto goes for headers. This will also probably make tests easier to write with webmock or similar.

Personally I would change
```
resp = @connection.get('/solr/admin/configs?action=LIST&omitHeader=true')
```

To
```
resp = @connection.get('/solr/admin/configs', {action: 'LIST', omitHeader: true})
```